www/js/pages/trade/components/purchase/purchase.controller.js

```diff
@@ -97,21 +100,32 @@
         });
 
         $scope.$on("purchase", (e, response) => {
+            const purchaseInfo = response.buy;
+
             if (!_.isEmpty(response.buy)) {
                 vm.showSummary = true;
                 $scope.$applyAsync(() => {
                     vm.purchasedContract = {
-                        contractId   : response.buy.contract_id,
-                        longcode     : response.buy.longcode,
+                        contractId   : purchaseInfo.contract_id,
+                        longcode     : purchaseInfo.longcode,
                         payout       : vm.proposalResponses[vm.purchasedContractIndex].payout,
-                        cost         : response.buy.buy_price,
-                        balance      : response.buy.balance_after,
-                        transactionId: response.buy.transaction_id,
+                        cost         : purchaseInfo.buy_price,
+                        balance      : purchaseInfo.balance_after,
+                        transactionId: purchaseInfo.transaction_id,
                         profit       : parseFloat(vm.proposalResponses[vm.purchasedContractIndex].payout) -
-                        parseFloat(response.buy.buy_price),
+                        parseFloat(purchaseInfo.buy_price),
                     };
                 });
-                websocketService.sendRequestFor.portfolio();
+                chartService.addContract({
```

Comment:
@nazaninreihani if you don't need the response of `ProposalOpenContract` so it doesn't need to subscribe to `buy` contract. As you know in the responses of `buy` subscription after the first response doesn't contain `buy` object because by subscribing to `buy`, it seems that you send a `buy` request and instantly subscribe to `ProposalOpenContract`.